### PR TITLE
feat(git): add neogit

### PR DESCRIPTION
**Pull request by taskylizard**

### CODE REVIEW

File: lua/astrocommunity/git/neogit/neogit.lua [hunk 1]
```diff
@@ -0,0 +1,16 @@
+return {
+  {
+    "TimUntersberger/neogit",
+    dependencies = {
+      { "nvim-lua/plenary.nvim" },
+    },
+    event = "User AstroGitFile",
+    keys = {
+      { "<leader>gn", desc = "Neogit", mode = { "n" } },
+      { "<leader>gnt", "<cmd>Neogit<CR>", desc = "Open Neoget Tab Page" },
```
Comment (mehalter) [inline lua/astrocommunity/git/neogit/neogit.lua:10]:
```suggestion
      { "<leader>gnt", "<cmd>Neogit<CR>", desc = "Open Neogit Tab Page" },
```

typo

File: lua/astrocommunity/git/neogit/neogit.lua [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      { "<leader>gnc", "<cmd>Neogit commit<CR>", desc = "Open Neogit Commit Page" },
+      { "<leader>gnd", ":Neogit cwd=", desc = "Open Neoget Override CWD" },
```
Comment (mehalter) [inline lua/astrocommunity/git/neogit/neogit.lua:12]:
```suggestion
      { "<leader>gnd", ":Neogit cwd=", desc = "Open Neogit Override CWD" },
```

typo

**Review by mehalter — CHANGES_REQUESTED** [review]
Looks great! just a few typos that need to be fixed and then we can get this merged right in

**Comment by taskylizard** [comment]
Resolved, please squash.

**Review by mehalter — APPROVED** [review]
LGTM

**Pull request merged**